### PR TITLE
Fix loading on Windows

### LINKED ISSUE #816: "DiskFS must use absolute paths" error when running all DT tests locally
I cloned the main DT repo and made some changes, then tried to run `pnpm run test-all`.  The result is the following error:

```
AssertionError [ERR_ASSERTION]: DiskFS must use absolute paths
    at DiskFS (C:\Workspace\DefinitelyTyped\node_modules\.pnpm\@definitelytyped+utils@0.0.184\node_modules\@definitelytyped\utils\src\fs.ts:175:11)
    at getDefinitelyTyped (C:\Workspace\DefinitelyTyped\node_modules\.pnpm\@definitelytyped+definitions-parser@0.0.189_typescript@5.4.0-dev.20231107\node_modules\@definitelytyped\definitions-parser\src\get-definitely-typed.ts:38:10)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at prepareAffectedPackages (C:\Workspace\DefinitelyTyped\node_modules\.pnpm\@definitelytyped+dtslint-runner@0.0.194_@octokit+core@5.0.1_typescript@5.4.0-dev.20231107\node_modules\@definitelytyped\dtslint-runner\src\prepareAffectedPackages.ts:16:14)
    at runDTSLint (C:\Workspace\DefinitelyTyped\node_modules\.pnpm\@definitelytyped+dtslint-runner@0.0.194_@octokit+core@5.0.1_typescript@5.4.0-dev.20231107\node_modules\@definitelytyped\dtslint-runner\src\main.ts:51:7)
    at <anonymous> (C:\Workspace\DefinitelyTyped\node_modules\.pnpm\@definitelytyped+dtslint-runner@0.0.194_@octokit+core@5.0.1_typescript@5.4.0-dev.20231107\node_modules\@definitelytyped\dtslint-runner\src\index.ts:111:22)
```

I haven't seen anybody else running into this, but I don't think I have a very unusual environment -- running directly in Git Bash on Git For Windows (Win11), installed Node 20.9.0 via `fnm`, then pnpm v8.9.0 via corepack.  Testing just the package I changed (`node`) via `pnpm run test node` works fine.

### PR DESCRIPTION
**Pull request by jakebailey**
Absolute paths on Windows don't start with `/`. Also, things get funky if we exec using a shell; use `execFile` + `which`.

Fixes #816

### CODE REVIEW

**Review by sandersn — APPROVED** [review]

**Comment by jakebailey** [comment]
Not quite there yet for Windows, apparently: 
image

Will look...

**Comment by jakebailey** [comment]
This is more invasive now; more path shennanegins but also it turns out that our command execution was not working properly either. In a followup I'd personally prefer to try and depend on `execa` instead.

**Comment by jakebailey** [comment]
It's too expensive to have it run everything, but it's possible we could have a CI job which only runs tests on one package just to make sure the flow works.

**Review by andrewbranch — APPROVED** [review]

File: packages/definitions-parser/src/get-affected-packages.ts [hunk 1]
```diff
@@ -18,9 +18,11 @@ export async function getAffectedPackages(
   const errors = [];
   // No ... prefix; we only want packages that were actually edited.
   const changedPackageDirectories = await execAndThrowErrors(
-    `pnpm ls -r --depth -1 --parseable --filter '@types/**[${sourceRemote}/${sourceBranch}]'`,
+    "pnpm",
+    ["ls", "-r", "--depth", "-1", "--parseable", "--filter", `@types/**[${sourceRemote}/${sourceBranch}]`],
     definitelyTypedPath
   );
+  console.log("changedPackageDirectories", changedPackageDirectories);
```
Comment (jakebailey) [inline packages/definitions-parser/src/get-affected-packages.ts:25]:
```suggestion
```

**Pull request merged**